core/routes/collection.ts

```diff
@@ -154,11 +154,23 @@ export default function (app: Hono) {
       const body = await c.req.parseBody();
       let name = body._id as string;
 
+      const namePrefix = "changes."
       if (!name && collection.nameField) {
-        const autoname = body[`changes.${collection.nameField}`];
-
-        if (typeof autoname === "string") {
-          name = autoname.replaceAll("/", "").trim();
+        if (typeof collection.nameField === "string") {
+          const autoname = body[`${namePrefix}${collection.nameField}`];
+
+          if (typeof autoname === "string") {
+            name = autoname.replaceAll("/", "").trim();
+          }
+        } else if (typeof collection.nameField === "function") {
+          const changedFields: Record<string, string> = {};
+          for (const key in body) {
+            if (key.startsWith(namePrefix) && typeof body[key] === "string") {
+              changedFields[key.substring(namePrefix.length)] = body[key];
+            }
+          }
+
+          name = collection.nameField(changedFields);
```

Comment:
Please, use the function `changesToData`, to convert the raw body to an object.

Reply:
much better!
The function is also called after creating the Document. Maybe we should call it only once and use the Data object in both places?

Reply:
I think you need to call `changesToData` each time because the body is created for each request.

By the way, I noticed that Hono has an option to parse Body into nested objects.
If you use it, you may not need `changesToData` at all

https://hono.dev/docs/api/request#dot-notation

```typescript
const body = await c.req.parseBody({ dot: true });
// [Object: null prototype] {
//   _id: "",
//   _prefix: "",
//   changes: [Object: null prototype] { title: "TITLE", description: "" }
// }
```

Reply:
I've open another PR: https://github.com/lumeland/cms/pull/27

Reply:
> I think you need to call changesToData each time because the body is created for each request.

What I mean is the `changesToData` function is called in line 183 to set the data to the new document. If it's called before the switch, the same object can be used by nameField and the new document.

About the Hono parse body option, I'm not sure it can be used. `changesToData` prevents numeric strings to be [ordered automatically](https://dev.to/frehner/the-order-of-js-object-keys-458d) and this is something that Hono's implementation doesn't do, but it's required by the CMS to allow to reorder list fields (list, object-list, etc).

Reply:
Oh, sorry. I misunderstood!
I've fixed to use the common variable after `changesToData`.